execution/tests/posix/execution.cc

```diff
@@ -726,3 +726,23 @@ SCENARIO("executing commands with execution::each_line") {
         }
     }
 }
+SCENARIO("Toggle expanding command paths behaviour in execution::expand_command") {
```

Comment:
I think we should need for cases for all combinations is builtin/command + expand true/false